### PR TITLE
add code to delete model cache caused by old training script

### PR DESCRIPTION
**Pull request by DreamGallery**

### CODE REVIEW

File: src/validate.py [hunk 1]
```diff
@@ -299,6 +302,11 @@ def loop(validation_args_file: str, task_id: str = None):
     last_successful_request_time = [time.time()] * len(task_id_list)
 
     while True:
+        for root, dirs, _ in os.walk(file_utils.default_cache_path):
```
Comment (nickcom007) [inline src/validate.py:305]: Will the `default_cache_path` still be accurate if someone has set `HF_HOME` to a different path?

**Review by nickcom007 — CHANGES_REQUESTED** [review]
I believe we should have a flag for user to determine whether they want the script to automatically do so. The default value can be `True`

File: src/validate.py (hunk 1 above)
Comment (DreamGallery) [reply src/validate.py:305]: yes, the code about [file_utils.default_cache_path](https://huggingface.co/transformers/v4.1.1/_modules/transformers/file_utils.html)
```
hf_cache_home = os.path.expanduser(
    os.getenv("HF_HOME", os.path.join(os.getenv("XDG_CACHE_HOME", "~/.cache"), "huggingface"))
)
default_cache_path = os.path.join(hf_cache_home, "transformers")
```

**Comment by DreamGallery** [comment]
> I believe we should have a flag for user to determine whether they want the script to automatically do so. The default value can be `True`

Add a flag `CLEAN_MODELS_CACHE` get from env to set whether to clear model cache, default value is `True`

File: src/validate.py [hunk 2]
```diff
@@ -36,6 +39,10 @@
     raise ValueError(
         "You need to set HF_TOKEN to download some gated model from HuggingFace"
     )
+CLEAN_MODELS_CACHE = os.getenv("CLEAN_MODELS_CACHE")
```
Comment (nickcom007) [inline src/validate.py:42]: Sorry I didn't make this clear enough previsouly. We should make this as an flag to `loop` command.
something like `--auto_clean_cache`

File: src/validate.py [hunk 3]
```diff
@@ -299,6 +306,12 @@ def loop(validation_args_file: str, task_id: str = None):
     last_successful_request_time = [time.time()] * len(task_id_list)
 
     while True:
+        if CLEAN_MODELS_CACHE:
+            for root, dirs, _ in os.walk(file_utils.default_cache_path):
```
Comment (nickcom007) [inline src/validate.py:310]: create a helper function for this to make it clean

Comment (DreamGallery) [reply src/validate.py]: done
use `click.option` add `--auto_clean_cache` and move clean cache code to a function `clean_model_cache`

File: src/validate.py [hunk 4]
```diff
@@ -158,6 +161,20 @@ def load_sft_dataset(
     return UnifiedSFTDataset(eval_file, tokenizer, max_seq_length, template)
 
 
+def clean_model_cache(auto_clean_cache: bool, cache_path: str = file_utils.default_cache_path):
```
Comment (nickcom007) [inline src/validate.py:164]:
```suggestion
def clean_model_cache(auto_clean_cache: bool, cache_path: str = file_utils.default_cache_path):
    """
    Cleans up the local model cache directory by removing directories that are not
    listed in SUPPORTED_BASE_MODELS.

    Parameters:
    - auto_clean_cache (bool): A flag to determine whether to clean the cache.
    - cache_path (str): The path to the cache directory. Defaults to file_utils.default_cache_path.
    """
    if not auto_clean_cache:
        return
    
    try:
        cache_path = Path(cache_path)
        for item in cache_path.iterdir():
            if item.is_dir() and item.name.startswith("models"):
                if item.name not in {f"models--{BASE_MODEL.replace('/', '--')}" for BASE_MODEL in SUPPORTED_BASE_MODELS}:
                    shutil.rmtree(item)
                    logger.info(f"Removed directory: {item}")
        logger.info("Successfully cleaned up the local model cache")
    except (OSError, shutil.Error) as e:
        logger.error(f"Failed to clean up the local model cache: {e}")
```

**Review by nickcom007 — CHANGES_REQUESTED** [review]
Recomend using `pathlib.Path` to make things cleaner. Please add the import as well

**Comment by DreamGallery** [comment]
Add the suggestion and resolved the conflicts, but I can‘t to rebase commits(total 4 commits logs😂).

**Comment by nickcom007** [comment]
LGTM now. Can you leave your ERC-20 address in the PR?

**Review by nickcom007 — APPROVED** [review]

**Comment by DreamGallery** [comment]
my flock trainer/validator address:
0xB5C318E67Fa21C0c7E27E64A6A54fBD03f709aE0
thank you

**Pull request merged**